cpmpy/solvers/gurobi.py

```diff
@@ -265,12 +265,12 @@ def transform(self, cpm_expr):
         # apply transformations, then post internally
         # expressions have to be linearized to fit in MIP model. See /transformations/linearize
         cpm_cons = flatten_constraint(cpm_expr)  # flat normal form
-        cpm_cons = decompose_global(cpm_cons, supported={"min","max", "element", "alldifferent"}) #element and alldiff have specialized MIP decomp in linearize
+        cpm_cons = decompose_global(cpm_cons, supported={"min","max","alldifferent"}) # alldiff has specialized MIP decomp in linearize
         cpm_cons = reify_rewrite(cpm_cons, supported=frozenset(['sum', 'wsum']))  # constraints that support reification
         cpm_cons = only_bv_implies(cpm_cons)  # anything that can create full reif should go above...
-        cpm_cons = linearize_constraint(cpm_cons)  # the core of the MIP-linearization
         cpm_cons = only_numexpr_equality(cpm_cons, supported=frozenset(["sum", "wsum", "sub"]))  # supports >, <, !=
-        cpm_cons = only_positive_bv(cpm_cons)  # after linearisation, rewrite ~bv into 1-bv
+        cpm_cons = linearize_constraint(cpm_cons, supported=frozenset({"sum", "wsum","sub","min","max","mul","abs","pow","div"}))  # the core of the MIP-linearization
```

Comment:
So `wsum, sum, sub, min, max, mul, div, abs` etc. remain after linearize? Is there a transformation that turns all of these into `wsum`s?

Reply:
For gurobi these indeed remain in the expressions.
`sub, min, max` (and soon `abs`)  will be linearized if they are not part of the supported set.
`mul` gets linearized to a `wsum` of one argument if the expression consists of a constant and a variable, otherwise we raise an error as we cannot linearize `var * var`.
For the `div` case, I am not sure how to linearize floordivision? If we can do so, we can cover the same case as mul. I.e., a `var / const`